catroid/src/main/java/org/catrobat/catroid/transfers/project/ProjectUploadService.kt

```diff
@@ -0,0 +1,206 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.transfers.project
+
+import android.app.IntentService
+import android.app.Notification
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.os.ResultReceiver
+import android.preference.PreferenceManager
+import android.support.v4.app.NotificationCompat
+import android.util.Log
+import org.catrobat.catroid.R
+import org.catrobat.catroid.common.Constants
+import org.catrobat.catroid.common.Constants.CATROBAT_EXTENSION
+import org.catrobat.catroid.common.Constants.EMAIL
+import org.catrobat.catroid.common.Constants.EXTRA_LANGUAGE
+import org.catrobat.catroid.common.Constants.EXTRA_PROJECT_DESCRIPTION
+import org.catrobat.catroid.common.Constants.EXTRA_PROJECT_NAME
+import org.catrobat.catroid.common.Constants.EXTRA_PROJECT_PATH
+import org.catrobat.catroid.common.Constants.EXTRA_PROVIDER
+import org.catrobat.catroid.common.Constants.EXTRA_RESULT_RECEIVER
+import org.catrobat.catroid.common.Constants.EXTRA_SCENE_NAMES
+import org.catrobat.catroid.common.Constants.EXTRA_UPLOAD_NAME
+import org.catrobat.catroid.common.Constants.EXTRA_USER_EMAIL
+import org.catrobat.catroid.common.Constants.FACEBOOK
+import org.catrobat.catroid.common.Constants.FACEBOOK_EMAIL
+import org.catrobat.catroid.common.Constants.GOOGLE_EMAIL
+import org.catrobat.catroid.common.Constants.GOOGLE_PLUS
+import org.catrobat.catroid.common.Constants.MAX_PERCENT
+import org.catrobat.catroid.common.Constants.NO_EMAIL
+import org.catrobat.catroid.common.Constants.NO_FACEBOOK_EMAIL
+import org.catrobat.catroid.common.Constants.NO_GOOGLE_EMAIL
+import org.catrobat.catroid.common.Constants.UPLOAD_RESULT_RECEIVER_RESULT_CODE
+import org.catrobat.catroid.io.ProjectAndSceneScreenshotLoader
+import org.catrobat.catroid.io.ZipArchiver
+import org.catrobat.catroid.ui.MainMenuActivity
+import org.catrobat.catroid.utils.DeviceSettingsProvider
+import org.catrobat.catroid.utils.ToastUtil
+import org.catrobat.catroid.utils.Utils
+import org.catrobat.catroid.utils.notifications.StatusBarNotificationManager
+import org.catrobat.catroid.utils.notifications.StatusBarNotificationManager.UPLOAD_NOTIFICATION_ID
+import org.catrobat.catroid.web.ServerCalls
+import java.io.File
+import java.util.Locale
+
+val UPLOAD_FILE_NAME = "upload$CATROBAT_EXTENSION"
+
+class ProjectUploadService : IntentService("ProjectUploadService") {
+
+    override fun onHandleIntent(projectUploadIntent: Intent?) {
+        val intent = projectUploadIntent ?: return
+
+        val projectPath = intent.getStringExtra(EXTRA_PROJECT_PATH) ?: return
```

Comment:
Hm, you've got a point here. I'd not throw an exception here, since it could cause the whole app to crash. Rather just log an error (which can easily be done by just returning a unit function which logs the error that some of the prerequisities have not been fulfilled)

Reply:
logging warning now in case of early return